test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,292 @@
+        "## Header 1.1\n"
+        "### Subheader 1.1.1\n"
+        "Content under sub-header 1.1.1\n"
+        "## Header 1.2\n"
+        "### Subheader 1.2.1\n"
+        "Content under header 1.2.1.\n"
+        "### Subheader 1.2.2\n"
+        "Content under header 1.2.2.\n"
+        "### Subheader 1.2.3\n"
+        "Content under header 1.2.3."
+    )
+
+
+# Basic splitting and structure
+def test_basic_split(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Should split into all headers with content
+    headers = [doc.meta["header"] for doc in split_docs]
+    assert "Header 1" in headers
+    assert "Subheader 1.1.1" in headers
+    assert "Subheader 1.2.1" in headers
+    assert "Subheader 1.2.2" in headers
+    assert "Subheader 1.2.3" in headers
+
+    # Check that content is present and correct
+    header1_doc = next(doc for doc in split_docs if doc.meta["header"] == "Header 1")
+    assert "Content under header 1." in header1_doc.content
+
+    subheader111_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.1.1")
+    assert "Content under sub-header 1.1.1" in subheader111_doc.content
+
+    subheader121_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.1")
+    assert "Content under header 1.2.1." in subheader121_doc.content
+
+    subheader122_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.2")
+    assert "Content under header 1.2.2." in subheader122_doc.content
+
+    subheader123_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.3")
+    assert "Content under header 1.2.3." in subheader123_doc.content
+
+    # Ensure all documents have a header in their metadata
+    for doc in split_docs:
+        assert doc.meta.get("header") is not None
+
+
+def test_split_parentheaders(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=sample_text), Document(content="# H1\n## H2\n### H3\nContent")]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    # Check parentheaders for both a deep subheader and a simple one
+    subheader_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.2")
+    assert "Header 1" in subheader_doc.meta["parent_headers"]
+    assert "Header 1.2" in subheader_doc.meta["parent_headers"]
+    h3_doc = next((doc for doc in split_docs if doc.meta["header"] == "H3"), None)
+    assert h3_doc.meta["parent_headers"] == ["H1", "H2"]
+
+
+def test_split_no_headers():
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content="No headers here."), Document(content="Just some text without headers.")]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    # Should return one doc per input, header is None
+    assert len(split_docs) == 2
```

Comment:
Just for sanity checking could we add 
```suggestion
    assert len(split_docs) == 2
    assert split_docs[0].content == docs[0].content
    assert split_docs[1].content == docs[1].content
```
testing that we return the same content